File: 卡码/代码随想录/Python/二叉树/02-二叉树属性/LeetCode_0100.py
#100. 相同的树
from typing import *
from ..TreeNode import *

class Solution:
  def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
    if p is not None and q is None:
      return False
    elif p is None and q is not None:
      return False
    elif p is None and q is None:
      return True
    elif p.val!=q.val:
      return False
    else:
      return self.isSameTree(p.left,q.left) and self.isSameTree(p.right,q.right)

File: 卡码/代码随想录/C++/二叉树/02-二叉树属性/LeetCode_0100.cpp
//100. ÏàÍ¬µÄÊ÷
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  bool isSameTree(TreeNode* p, TreeNode* q) {
		if(p==nullptr&&q!=nullptr) return false;
		else if(p!=nullptr&&q==nullptr) return false;
		else if(p==nullptr&&q==nullptr) return true;
		else if(p->val!=q->val) return false;
		else return isSameTree(p->left,q->left)&&isSameTree(p->right,q->right);
  }
};

int main(){
  system("pause");
  return 0;
}
